pkg/didmethod/peer/store.go

```diff
@@ -0,0 +1,117 @@
+/*
+Copyright SecureKey Technologies Inc. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package peer
+
+import (
+	"encoding/base64"
+	"encoding/json"
+	"time"
+
+	"github.com/hyperledger/aries-framework-go/pkg/doc/did"
+	"github.com/hyperledger/aries-framework-go/pkg/storage"
+	"golang.org/x/xerrors"
+)
+
+// DIDModifiedBy key/signature used to update the Peer DID Document
+type DIDModifiedBy struct {
+	Key string `json:"key,omitempty"`
+	Sig string `json:"sig,omitempty"`
+}
+
+type docDelta struct {
+	Change     string           `json:"change,omitempty"`
+	ModifiedBy *[]DIDModifiedBy `json:"by,omitempty"`
+	ModifiedAt time.Time        `json:"when,omitempty"`
+}
+
+// DIDStore Peer DID Document store
+type DIDStore struct {
+	store storage.Store
+}
+
+// NewDIDStore new Peer DID store (backing store is configurable)
+func NewDIDStore(s storage.Store) *DIDStore {
+	return &DIDStore{
+		store: s,
+	}
+}
+
+// Put saves Peer DID Document along with user key/signature
+func (s *DIDStore) Put(did string, doc *did.Doc, by *[]DIDModifiedBy) error {
+	if did == "" || doc == nil {
+		return xerrors.New("DID and Document are mandatory")
+	}
+
+	var deltas []docDelta
+
+	// TODO - Need to derive the docDelta if its not a genesis document(DID already exists)
+	// (https://github.com/hyperledger/aries-framework-go/issues/54)
+	// For now, assume the doc is a genesis document
+	jsonDoc, err := json.Marshal(doc)
+	if err != nil {
+		return xerrors.Errorf("error marshalling the DID document: %w", err)
```

Comment:
nitpick: i usually like to put the failed (error) word at the end of the phrase (since we can guess it was an error)

Reply:
@troyronda updated the error message - https://github.com/hyperledger/aries-framework-go/compare/833a488e7fe5b8ef98198737e78a862440bb98a8..dcbe51e771012c337063cf3445d94e8d42b54c6b